### PR TITLE
Separate manual tests from CI tests

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
- Manual tests should not be triggered by CI.

#### Change overview
- Creates a separate list in chip-tool for manual tests. 
- Generates ZAP code.

#### Testing
- Generated code
- Compiled code
- Ran `chip-tool tests list`
- Ran `chip-tool tests list-manual`

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]
Could you move these in there too?
```
    'TestGroupDemoCommand',
    'TestGroupDemoConfig',
```
Those are also manual tests.

**Review by andy31415 — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: examples/chip-tool/templates/tests.js [hunk 1]
```diff
@@ -15,6 +15,29 @@
...
+  const DeviceDiscovery = [
+    'Test_TC_DD_1_5',
+    'Test_TC_DD_1_6',
+    'Test_TC_DD_1_7',
+    'Test_TC_DD_1_8',
+    'Test_TC_DD_1_9',
+  ];
+
+  const Groups = [
+    'TestGroupDemoCommand',
```
Comment (bzbarsky-apple) [inline examples/chip-tool/templates/tests.js:29]: Fwiw, as a followup we can remove the manual exclusion of these in `.github/workflows/tests.yaml`.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/templates/tests.js (hunk 1 above)
Comment (krypton36) [reply examples/chip-tool/templates/tests.js:29]: Ok. I created an issue [here](https://github.com/project-chip/connectedhomeip/issues/14753)

**Review by tcarmelveilleux — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking testing improvements.

**Pull request merged**